client/platform/desktop/backend/native/common.ts

```diff
@@ -604,6 +604,30 @@ async function _initializeProjectDir(settings: Settings, jsonMeta: JsonMeta): Pr
   return projectDir;
 }
 
+async function deleteDataset(
+  settings: Settings,
+  datasetId: string,
+): Promise<boolean> {
+  // confirm dataset Id exists
+  const projectDirInfo = await getValidatedProjectDir(settings, datasetId);
+  fs.rmdirSync(projectDirInfo.basePath, { recursive: true });
```

Comment:
Don't use synchronous fs methods.   `await fs.remove(projectDirInfo.basePath)` is an async recursive equivalent.

https://github.com/jprichardson/node-fs-extra/blob/master/docs/remove.md

Reply:
I don't know why I did a synchronous delete there but this brings up another question I had.  I feel very awkward about implementing a recursive delete on a user's file system based on a folder.  Should I be more explicit in the files I am deleting from the `projectDir` instead of just doing a recursive delete on the whole folder?